src/lib/functions/registry.js

```diff
@@ -113,7 +113,18 @@ class FunctionsRegistry {
     // If there is no watcher for this function but the build produced files,
     // we create a new watcher and watch them.
     if (srcFilesDiff.added.size !== 0) {
-      const newWatcher = await watchDebounced([...srcFilesDiff.added], {
+      const config = func.config.functions
+      const filesToWatch = [
+        ...srcFilesDiff.added,
+        // Include the manually specified files from config. These files might not be
+        // source files, but might still be used/loaded in the function and therefore
+        // we need to watch them to rebuild once they change
+        ...((config && config[func.name] && config[func.name].included_files) || []).filter(
```

Comment:
Hmm, like you said I don't love the fact that we're duplicating the logic we have in zip-it-and-ship-it. Maybe the data coming back from [`listFunctions`](https://github.com/netlify/cli/blob/eb3f87d3dc842388258d580cff04b7cc9dacb87a/src/lib/functions/registry.js#L186-L192) already contains the resolved list of files? If not, perhaps we could add a new property with that information?